### PR TITLE
Admissions find/replace 2024

### LINKED ISSUE #8316: One-off find and replace for Admissions
# Problem

Until we develop a find/replace solution (#4258), Admissions has requested a one-off update hook to update 100+ instances of a string found on their Area of Study content.

**Search**:
`Iowa graduates have a 95 percent job/grad school placement rate`
**Replace**:
`Iowa graduates have a 96 percent job/grad school placement rate`
**Scope**: Area of Study content type

# Additional info

We have a similar update hook in admissions_core (admissions_core_update_9001) for updating a link.

# Proposed Solution

- Use find text to see if we can scope it down to a particular set of fields.
- Write an update hook in admissions_core to loop through AoS content and update instances with the new placement rate saving as a new revision but also accounting for tempstore and drafts. No need to update all revisions.
- Do a test run in DEV so that Admissions staff can review the result.
- Backup the database in production before triggering the update.

### PR DESCRIPTION
**Pull request by joewhitsitt**
Resolves https://github.com/uiowa/uiowa/issues/8316

Leveraging the `DOMXPath` strategy I found in P2LbHelper and elsewhere in the codebase.

# How to test

- `git checkout main`
- `ddev blt ds --site=admissions.uiowa.edu && ddev drush @admissions.local uli admin/find-text`
- Search for `Iowa graduates have a 95 percent job/grad school placement rate` without render markup checked.
- Notice all of the instances come from the `Area Of Study Career` field.
- `git checkout admissions_find_replace_2024`
- `ddev drush @admissions.local updb` - notice no errors. should output the nodes that were changed.
- Search the string again. See there are no results.
- Search for `Iowa graduates have a 96 percent job/grad school placement rate` and see results.
- Check one of the recent AoS node saves. See the revision entry and the diff between versions.
- Go to https://admissions.uiowa.ddev.site/student-profiles and see images for the profiles along with additional person meta info.

### CODE REVIEW

File: docroot/sites/admissions.uiowa.edu/modules/admissions_core/src/Entity/StudentProfile.php
Comment (joewhitsitt) [inline docroot/sites/admissions.uiowa.edu/modules/admissions_core/src/Entity/StudentProfile.php:1]: @bspeare do you recall the reasoning of the change to this file? https://github.com/uiowa/uiowa/pull/8195

Comment (bspeare) [reply docroot/sites/admissions.uiowa.edu/modules/admissions_core/src/Entity/StudentProfile.php:1]: No reason, it looks like a mistake.

**Comment by joewhitsitt** [comment]
Going to deploy this out to DEV for testing and review 12/11.

**Comment by joewhitsitt** [comment]
Got approval from Admissions to move forward with this pending developer approval

**Review by cory-skeers — APPROVED** [review]
Reviewed and tested, and everything worked great. Confirmed from the terminal output that the number of nodes updated matched the number of results from the original search string.

I think there may be some manual cleanup in addition to this doing a search for "95 percent" after the hook.

`https://admissions.uiowa.edu/academics/management`: "Iowa's graduates have a 95 percent job/grad school placement rate within six months of graduation." (extra 's so it doesn't get seen by the update hook)
`https://admissions.uiowa.edu/node/3556`  "Iowa graduates has a 95 percent job/grad school placement rate within six months of graduation." ('has a')
`https://admissions.uiowa.edu/node/3351/` "Iowa graduates have a 95 percent job/grad school&nbsp;placement rate within six months of graduation." (`&nbsp` not seen by the search) <- couple of similar instances of these.

Overall it's 7 nodes, and I think would be fine with manual updates since this still covers the other ~100 instances.

**Comment by joewhitsitt** [comment]
I wonder if they would be open to correcting these ahead of deployment?

**Comment by cory-skeers** [comment]
Update: 6 nodes not 7, and two of them are along the lines of "Iowa graduates have a 95 percent job placement rate within six months of graduation" (`https://admissions.uiowa.edu/node/4396/`, `https://admissions.uiowa.edu/node/3936/`) which isn't seen because of the "job" rather than "job/grad" -- I'm assuming that should be updated as well, but there's a chance that's intentional.

**Comment by cory-skeers** [comment]
> I wonder if they would be open to correcting these ahead of deployment?

Full nodelist:

- 2846
- 3351
- 3556
- 3691
- 3936
- 4396

**Comment by joewhitsitt** [comment]
Thanks @cory-skeers. I can pass this along to them to see if they are up to it.

**Comment by joewhitsitt** [comment]
>Yes, I think we encountered the same thing and made a list of ones to hit manually. - Scott

**Comment by joewhitsitt** [comment]
@briand44 I told them I would let them know when this was deployed but I might be out when it does.

**Pull request merged**